tls/s2n_internal.h

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#pragma once
+
+#if ((__GNUC__ >= 4) || defined(__clang__)) && defined(S2N_EXPORTS)
+#    define S2N_API __attribute__((visibility("default")))
+#else
+#    define S2N_API
+#endif /* __GNUC__ >= 4 || defined(__clang__) */
```

Comment:
Rather than redefine this the same, I'd suggest either:
* Defining an S2N_PRIVATE_API, so that the name is at least helpful and we COULD add more build restrictions later
* Moving the original definition to an s2n_api.h header that can be included both here and in s2n.h

Reply:
i like the S2N_PRIVATE_API idea because it conveys the intent and also will let us iterate on visibility